jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/io/WebSocketCoreConnection.java

```diff
@@ -122,8 +122,8 @@ public WebSocketCoreConnection(EndPoint endp, Executor executor, ByteBufferPool
         Objects.requireNonNull(decoratedObjectFactory, "DecoratedObjectFactory");
         Objects.requireNonNull(policy, "WSPolicy");
         Objects.requireNonNull(extensionStack, "ExtensionStack");
-        Objects.requireNonNull(upgradeRequest, "UpgradeRequest");
-        Objects.requireNonNull(upgradeResponse, "UpgradeResponse");
+        // TODO Objects.requireNonNull(upgradeRequest, "UpgradeRequest");
+        // TODO Objects.requireNonNull(upgradeResponse, "UpgradeResponse");
```

Comment:
This is important, upgrade request and response are required.

Reply:
They are required by some API layers, but I can't see how they are needed by core.  In fact core is working fine with them null!
I understand that the abstraction of the requests needs to be carried by the session, but that can be added in the API layer as either a session wrapper or extension in a way that is specific to the API layer.